src/library_fs.js

```diff
@@ -895,10 +902,7 @@ FS.staticInit();
     readdir(path) {
       var lookup = FS.lookupPath(path, { follow: true });
       var node = lookup.node;
-      if (!node.node_ops.readdir) {
-        throw new FS.ErrnoError({{{ cDefs.ENOTDIR }}});
-      }
-      return node.node_ops.readdir(node);
+      return FS.doNodeOp(node, "readdir", {{{ cDefs.ENOTDIR }}});
```

Comment:
When I first say this change I assumed `Node` here meant  Node JS.. but I just the term just very overloaded.

I worry a little that using strings here will cause closure to break, or at least fail to minify these attribute names.

How about this `return FS.doNodeOp(node.node_ops.readdir, {{{ cDefs.ENOTDIR }}});` .. thus avoid the string lookup and passing instead the result of the lookup?

Reply:
Or maybe we can leave the call at the callsite and just make check function that doesn't need to take the `...` operator (which I believe generates JS garbage (i.e. a new array)).

```
FS.checkExists(node.readdir, {{{ cDefs.ENOTDIR }}});
```

Reply:
Well then we'd have to pass both `node` and `node.node_ops.readdir`.

Reply:
Still, same number of args that you have today, but avoids the extra strings + lambdas I think?

Reply:
Okay, I switched to checkExists as you suggested.